tools/common/StaticRegistrar.cs

```diff
@@ -3690,6 +3677,14 @@ void Generate (string header_path, string source_path)
 							hdr.WriteLine ("#include <objc/runtime.h>");
 							hdr.WriteLine ("#include <objc/message.h>");
 
+#if MONOMAC
+							// 32-bit header files do not define these types
+							if (!Is64Bits) {
```

Comment:
I think it makes more sense to check the availability attribute and see if each type is 64bit only instead of hardcoding specific types like this.